nncf/common/accuracy_aware_training/training_loop.py

```diff
@@ -24,20 +24,36 @@
 from nncf.common.utils.logger import logger as nncf_logger
 from nncf.common.utils.registry import Registry
 from nncf.config.config import NNCFConfig
-from nncf.common.utils.backend import __nncf_backend__
-
-if __nncf_backend__ == 'Torch':
-    from nncf.torch.accuracy_aware_training.runner import PTAccuracyAwareTrainingRunner as \
-        AccuracyAwareTrainingRunner
-elif __nncf_backend__ == 'TensorFlow':
-    from nncf.tensorflow.accuracy_aware_training.runner import TFAccuracyAwareTrainingRunner as \
-        AccuracyAwareTrainingRunner
 
 
 ModelType = TypeVar('ModelType')
 ADAPTIVE_COMPRESSION_CONTROLLERS = Registry('adaptive_compression_controllers')
 
 
+def get_backend_specific_training_runner_cls(compression_controller: CompressionAlgorithmController,
+                                             nncf_backend: str = None):
+    try:
+        import torch
+    except ImportError:
+        torch = None
+
+    try:
+        import tensorflow
+    except ImportError:
+        tensorflow = None
+
+    if torch is not None and isinstance(compression_controller.model, torch.nn.Module):
+        from nncf.torch.accuracy_aware_training.runner import PTAccuracyAwareTrainingRunner
+        return PTAccuracyAwareTrainingRunner
+
+    if tensorflow is not None and isinstance(compression_controller.model, tensorflow.Module):
```

Comment:
@alexsu52, can you please look at this part specifically and comment if it is valid?

Reply:
@AlexKoff88, yes we discussed this with @vanyalzr. I can confirm that it is valid.